### PR TITLE
Use clang as assembler to ensure preprocessing is done, and add assembler "target" flag.

### LINKED ISSUE #62810: Compiling for Android on Windows fails
### Godot version

master (f60cb2931c74d68b19b18c15442c1043d2ebb884)

### System information

Windows 11 21H2, Android

### Issue description

Building for Android on Windows currently fails. This is a regression from 39ed39900e34e10f40d0d06ed358d2641d7f3315 (#59167), which added [thirdparty/zstd/decompress/huf_decompress_amd64.S](https://github.com/godotengine/godot/commit/39ed39900e34e10f40d0d06ed358d2641d7f3315#diff-5e087c5e7e789af198d41e864c0912fa47c4158c024879fa253a8ba1b4f692e2).

With `android_arch=armv7`:
```
C:\Android\android-sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/arm-linux-androideabi/bin/as -o thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.armv7.o thirdparty\zstd\decompress\huf_decompress_amd64.S
=====
scons: *** [thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.armv7.o] Error 1
b"thirdparty\\zstd\\decompress\\huf_decompress_amd64.S: Assembler messages:\r\nthirdparty\\zstd\\decompress\\huf_decompress_amd64.S:33: Error: bad instruction `zstd_hide_asm_function(HUF_decompress4X1_usingDTable_internal_bmi2_asm_loop)'\r\nthirdparty\\zstd\\decompress
```
With `android_arch=arm64v8`:
```
C:\Android\android-sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/aarch64-linux-android/bin/as -o thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.armv8.o thirdparty\zstd\decompress\huf_decompress_amd64.S
=====
 scons: *** [thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.armv8.o] Error 1
b"thirdparty\\zstd\\decompress\\huf_decompress_amd64.S: Assembler messages:\r\nthirdparty\\zstd\\decompress\\huf_decompress_amd64.S:33: Error: unknown mnemonic `zstd_hide_asm_function' -- `zstd_hide_asm_function(HUF_decompress4X1_usingDTable_internal_bmi2_asm_loop)'\r\nthirdparty\\zstd\\decompress
```
With `android_arch=x86`
```
C:\Android\android-sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/i686-linux-android/bin/as -o thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.x86.o thirdparty\zstd\decompress\huf_decompress_amd64.S
=====
scons: *** [thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.x86.o] Error 1
b"thirdparty\\zstd\\decompress\\huf_decompress_amd64.S: Assembler messages:\r\nthirdparty\\zstd\\decompress\\huf_decompress_amd64.S:33: Error: no such instruction: `zstd_hide_asm_function(HUF_decompress4X1_usingDTable_internal_bmi2_asm_loop)'\r\nthirdparty\\zstd\\decompress
```
With `android_arch=x86_64`
```
C:\Android\android-sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/x86_64-linux-android/bin/as -o thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.x86_64.o thirdparty\zstd\decompress\huf_decompress_amd64.S
=====
scons: *** [thirdparty\zstd\decompress\huf_decompress_amd64.android.opt.x86_64.o] Error 1
b"thirdparty\\zstd\\decompress\\huf_decompress_amd64.S: Assembler messages:\r\nthirdparty\\zstd\\decompress\\huf_decompress_amd64.S:33: Error: no such instruction: `zstd_hide_asm_function(HUF_decompress4X1_usingDTable_internal_bmi2_asm_loop)'\r\nthirdparty\\zstd\\decompress
```
Note: Each architecture uses it's own `as` assembler compiler; so there is no need to specify the architecture as done in #62587 (the fix for Macos).

### Steps to reproduce

On Windows on the master branch run:
- `scons platform=android android_arch=armv7`
- `scons platform=android android_arch=arm64v8`
- `scons platform=android android_arch=x86`
- `scons platform=android android_arch=x86_64`
 

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by bruvzg**
Supersede #62811
Fixes #62810

### CODE REVIEW

**Comment by bruvzg** [comment]
Seems to be building fine on Windows, Linux and macOS with `arm64v8`, `armv7` and `x86_64` Android archs.

File: platform/android/detect.py [hunk 1]
```diff
@@ -158,7 +159,7 @@ def configure(env):
     env["CXX"] = compiler_path + "/clang++"
     env["AR"] = compiler_path + "/llvm-ar"
     env["RANLIB"] = compiler_path + "/llvm-ranlib"
-    env["AS"] = bin_utils_path + "/as"
+    env["AS"] = compiler_path + "/clang"
```
Comment (akien-mga) [inline platform/android/detect.py:162]: `bin_utils` and `bin_utils_path` could be removed, they were only defined for this.

**Review by akien-mga — APPROVED** [review]

**Comment by madmiraal** [comment]
See build log: https://github.com/madmiraal/godot/actions/runs/2630314597

**Comment by bruvzg** [comment]
> this change wouldn't work on 3.x as it breaks the libvpx assembly build (which IMO might itself be badly configured, the libvpx SCsub is a mess). Will need looking into if/when cherry-picking the zstd update.

The following patch seems to fix `libvpx` build for Android after backporting this PR to `3.x` (tested on Windows, with `armv7`, `arm64v8` and `x86_64` export template builds):

```diff
diff --git a/modules/webm/libvpx/SCsub b/modules/webm/libvpx/SCsub
index 8a4a6fb5f3..46bd7bf652 100644
--- a/modules/webm/libvpx/SCsub
+++ b/modules/webm/libvpx/SCsub
@@ -312,7 +312,7 @@ if webm_cpu_arm:
         or env["platform"] == "x11"
         or env["platform"] == "server"
     ):
-        env_libvpx["ASFLAGS"] = "-mfpu=neon"
+        env_libvpx["ASFLAGS"] += "-mfpu=neon"
     elif env["platform"] == "uwp":
         env_libvpx["AS"] = "armasm"
         env_libvpx["ASFLAGS"] = ""
```

**Comment by akien-mga** [comment]
Cherry-picked for 3.6. (With the diff in https://github.com/godotengine/godot/pull/62813#issuecomment-1179546418.)

**Pull request merged**